core/src/main/java/tachyon/worker/WorkerStorage.java

```diff
@@ -765,26 +825,64 @@ public void register() {
    * 
    * @param userId The id of the user who send the request
    * @param requestBytes The requested space size, in bytes
-   * @return true if succeed, false otherwise
+   * @return StorageDir assigned if success, null otherwise
    */
-  public boolean requestSpace(long userId, long requestBytes) {
-    LOG.info("requestSpace(" + userId + ", " + requestBytes + "): Current available: "
-        + mSpaceCounter.getAvailableBytes() + " requested: " + requestBytes);
-    if (mSpaceCounter.getCapacityBytes() < requestBytes) {
-      LOG.info("user_requestSpace(): requested memory size is larger than the total memory on"
-          + " the machine.");
-      return false;
+  private StorageDir requestSpace(StorageDir dirCandidate, long userId, long requestBytes) {
```

Comment:
Fix the javadoc comments of this method.
By the way, there are several `requestSpace` methods in WorkerStorage, StorageDir and StorageTier. It's more clearly to add comments to describe the workflow. E.g. which is used to create a block, and which is used to append the block.